bids/layout/index.py

```diff
@@ -164,10 +165,21 @@ def index_files(self):
         """Index all files in the BIDS dataset. """
         self._index_dir(self.root, self.config)
 
-    def index_metadata(self):
+    def index_metadata(self, **filters):
         """Index metadata for all files in the BIDS dataset. """
+        if filters:
+            # ensure we are returning objects
+            filters['return_type'] = 'object'
+            # until 0.11.0, user can specify extension or extensions
+            ext_key = 'extensions' if 'extensions' in filters else 'extension'
```

Comment:
`.get()` will handle the extensions/extension thing for you, so you don't need to worry about it here.

Reply:
Yeah, he only does because he has to add stuff to the `extensions` argument, and its not clear which one the user will pass.